IO-pc-local/src/app/lobby/lobby.service.ts

```diff
@@ -1,26 +1,43 @@
 import { Injectable } from '@angular/core';
-import { BehaviorSubject, Observable } from 'rxjs';
+import { Subject, Observable } from 'rxjs';
 import { Student } from './shared/model';
 
 @Injectable({
   providedIn: 'root'
 })
 export class LobbyService {
-  private readonly _students: BehaviorSubject<Student[]> = new BehaviorSubject([] as Student[]);
-  public students$: Observable<Student[]> = this._students.asObservable();
+  public readonly students = new Subject<Student[]>();
   private sse: EventSource | null = null;
 
+
   constructor() {
-    // this.sse = new EventSource("http://localhost:8000/", 
-    // { withCredentials: true, });
-    this._students.next(
-      [{name: "Maciuś"}, {name: "Zuzia"}, {name: "Franek"}, {name: "Janek"}]
-    )
+    this.sse = new EventSource("http://localhost:8000/game/lobby", 
+    { withCredentials: true, });
+
+    this.sse.onopen = open => {
+      console.log("open: ", open);
+    }
+
+    this.sse.onmessage = msg => {
+      console.log(msg.data);
+      this.students.next(JSON.parse(`${msg.data}`))
```

Comment:
Chyba zamiast `` `${msg.data}` `` mogłoby być po prostu `msg.data`

Reply:
To jest chyba pozostałość po mojej próbie naprawienia błędu z niepoprawnym JSON-em przesyłanym przez serwer. Ostatecznie naprawiłem to po stronie backendu także masz rację